ros_ws/src/car_control/include/drive_parameters_source.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <ros/ros.h>
+
+#include <drive_msgs/drive_param.h>
+#include <functional>
+#include <std_msgs/Float64.h>
+
+class DriveParametersSource;
+typedef std::function<void(DriveParametersSource*, const drive_msgs::drive_param::ConstPtr&)>
+    DriveParameterCallbackFunction;
+
+constexpr double IDLE_RANGE = 0.01;
+
+/*
+*  A class that listens on a topic that publishes drive parameters and stores information about that source
+*/
+class DriveParametersSource
+{
+    public:
+    /**
+     * @brief Construct a new DriveParametersSource object,
+     * subscribes to the topic and stores the parameters into local variables.
+     *
+     * @param node_handle A node handle is needed to create a topic subscriber
+     * @param topic The name of the drive_param topic to subscribe to
+     * @param update_callback Callback function that will be called whenever a message was received
+     * @param priority Priority of the source. Sources with higher priority will be preferred.
+     * @param timeout Messages will be deferred when they are older than this, in seconds.
+     */
+    DriveParametersSource(ros::NodeHandle* node_handle, const char* topic,
+                          DriveParameterCallbackFunction update_callback, int priority, double timeout);
+
+    /**
+     * @brief Returns true if no update was received for a certain time, determined by the timeout variable.
+     */
+    bool isOutdated();
+
+    /**
+     * @brief Returns true if the last update contained values close to 0 for both steering and throttle.
+     * If no update was received yet, this returns true.
+     */
+    bool isIdle();
+
+    int getPriority();
+
+    private:
+    ros::Subscriber m_drive_parameters_subscriber;
+
+    int m_priority;
+    long int m_timeout;
+    bool m_idle;
+    long int m_last_update;
```

Comment:
You should replace this POSIX time handling with `std::chrono` time handling, as the POSIX handling is neither portable nor does it account for weird stuff that happens to time e.g. switch to daylight saving time or time corrections via NTP. This would mean replacing the type `long int` of `m_last_update` with `std::chrono::time_point<T>`[1]. And `long int` of `m_timeout` with a duration `std::chrono::time_point<T>::duration`.
The member function `getTime` would then became obsolete as it can be replace with the STL function `std::chrono::time_point<T>::now()`.

Some other points as `DriveParametersSource::DriveParametersSource` would also needed to be changed.
```C++
// Old 
// [..]
this->m_timeout = (long int)(timeout * 1000.0);
this->m_last_update = 0;
// [..]

// New
// [..]
auto timeout_as_duration = std::chrono::duration<double>(timeout);
this->m_timeout = std::chrono::duration_cast<typeof(m_timeout)>(timeout_as_duration);
this->m_last_update = std::chrono::T::time_point::min();
// [..]
```
See also:
https://en.cppreference.com/w/cpp/chrono/time_point/min

[1]
T can be:
https://en.cppreference.com/w/cpp/chrono/system_clock
https://en.cppreference.com/w/cpp/chrono/steady_clock
https://en.cppreference.com/w/cpp/chrono/high_resolution_clock
I'm not sure which kind of clock would be best in the scenario, but I would assume `steady_clock`